### DISCUSSION TITLE
Bulk-ignoring faces in a directory and all its subdirectories

### DISCUSSION THREAD

**Discussion by ianfixes** [photoprism/photoprism#3973] [Q&A]
In addition to my personal photos, I'm using Photoprism to organize a large collection of art and photography -- portraits, street photos, photojournalism, etc.  The sheer number of faces recognized in this set seems to have overwhelmed photoprism's ability to find & tag my actual friends and family.  

It's an awful lot of clicking to individually ignore all the random strangers' photos in this subdirectory, and there are also a lot of false-positives where the random strangers are grouped with faces of people I actually know. (also, Abraham Lincoln pops up a lot between paintings and statues).  

Is there an SQL command I can run to both untag and ignore all the detected faces if I know the top-level directory name?  Or did I miss a configuration option to prune only part of the facial recognition tree?

**Comment by graciousgrey** [comment]
This is not possible at this time. Facial recognition can be disabled but this setting applies to the whole library. For your use case it might make sense to set up a separate instance for your art and photography collection.

**Reply by ianfixes** [reply]
Can you describe where the facial recognition tagging information is currently stored, and what change takes place in storage when I ignore a face via the GUI? I was expecting you to say "it's difficult" or "we haven't really looked at that", but not "it's not possible".

**Reply by ianfixes** [reply]
To be clear, I'd be very comfortable puzzling out the SQL command to run if someone can point me to the right tables and how to get an SQL console.  Worst case I just delete and reindex.

**Reply by graciousgrey** [reply]
If you [remove a face via the GUI](https://docs.photoprism.app/user-guide/organize/people/#remove-faces), the marker_invalid field in the markers table is set to 1. You can find the database schema [here](https://docs.photoprism.app/developer-guide/database/schema/).  

You could try to do this with an SQL query for all markers of files that have a specific path in their file name. However, unfortunately we cannot guarantee you that there will be no unwanted side effects on facial recognition, especially if you set markers to invalid that have a subject associated with them.

**Reply by lastzero** [reply]
The related DB tables are faces and markers, see https://docs.photoprism.app/developer-guide/database/schema/

Running photoprism faces help in a terminal will additionally show  CLI commands you can use.

**Reply by ianfixes** [reply]
Thank you.  It sounds like I need to disassociate subjects from markers prior to setting the markers to invalid.  

For a marker to be "invalid", does that imply that "the marker isn't a face" or just that "I don't care about the face that is marked"?

**Reply by lastzero** [reply]
By setting the `invalid` flag, the face marker will be ignored going forward without changing the marker type as such. However, you should also delete the related rows in the `faces` table based on the `face_id`:
https://github.com/photoprism/photoprism/blob/f8e0615cc86f9c9eb0e8276d0995f74e6996dc78/internal/entity/marker.go#L38

The `file_uid` and the filename can be found in the `files` table.
